data/transactions/logic/assembler_test.go

```diff
@@ -2233,6 +2233,28 @@ func TestAssembleConstants(t *testing.T) {
 	}
 }
 
+func TestIntAndIntcBlock(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	// Assembler should use pushint on int pseudo-op after program manually created intcblock unless int is already in the block
+	ops := testProg(t, "intcblock 0x01 0x02; int 0x04", AssemblerMaxVersion)
+	expected := []byte{byte(AssemblerMaxVersion), 0x20, 0x02, 0x01, 0x02, 0x81, 0x04}
+	require.Equal(t, expected, ops.Program)
+	// If the int is already in the block, we mind as well use the block
+	ops = testProg(t, "intcblock 0x01 0x02; int 0x02", AssemblerMaxVersion)
+	expected = []byte{byte(AssemblerMaxVersion), 0x20, 0x02, 0x01, 0x02, 0x23}
+	require.Equal(t, expected, ops.Program)
+	// Previous int pseudo-ops should be pushints
```

Comment:
We probably have a tough time really knowing what's "previous" here, right?  In this example it's correct, but with branching, you could have some code that appears before other code, but runs later.

Reply:
I agree, but it's also pretty nonsensical to only have a single intcblock but then also not put it at the beginning (except for some potential dynamic cost saving I suppose). I'm happy to change it to just if you manually put in a const block, the pseudo-ops all go to pushint/pushbytes, but this means that people who want to manually set up const blocks now also have to manually index them.

Another option could be to require the const block setups to be the first ops or else the pseudo-ops just go to pushint/pushbytes